dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java

```diff
@@ -10,346 +13,198 @@
-								statement = connection.createStatement();
-								statement.execute(commit);
-								task.executeUpgrade();
-								statement = connection.createStatement();
-								statement.execute(lock);
-							}else{
-							  task.executeUpgrade();
-							}
-						} 
-						// Nothing to execute, or the task ran ok so bump
-						// the db version.
-						try {
-//						    conn = DbConnectionFactory.getDataSource().getConnection();
-							if (connection != null && connection.isClosed()) {
-								connection = DbConnectionFactory.getDataSource().getConnection();
-							}
-						    connection.setAutoCommit(true);
-						    update = connection.prepareStatement("INSERT INTO db_version (db_version,date_update) VALUES (?,?)");
-    						update.setInt(1, taskId);
-    						Date date = new Date(Calendar.getInstance().getTimeInMillis());
-    						update.setDate(2, date);
-    						update.execute();
-						}
-						finally {
-//							statement.execute(commit);
-						    update.close();
-//						    conn.close();
-						}
-
-						Logger.info(this, "Database upgraded to version: "
-								+ taskId);
-						HibernateUtil.closeAndCommitTransaction();
-
-					}
-				} catch (NumberFormatException e) {
-					Logger
-							.error(
-									this,
-									"Class "
-											+ name
-											+ " has invalid name or shouldn't be in the tasks package.");
-
-				}
-			}
-	
-			ReindexThread.unpause();
-			
-		} catch (Throwable e) {
-			HibernateUtil.rollbackTransaction();
-      if(Config.getBooleanProperty("SYSTEM_EXIT_ON_STARTUP_FAILURE", true)){
-        e.printStackTrace();
-        System.exit(1);
-      }
-			throw new DotDataException("Unable to execute startup task : ",e);
-		} finally {
-			// This will commit the changes and close the connection
-			HibernateUtil.closeSession();
-			
-			// DOTCMS-4352
-			try {
-				if(connection != null && !connection.isClosed()){
-					Statement s1 = connection.createStatement();
-					if(DbConnectionFactory.isMySql()){
-						s1.execute(commit);
-					}
-					if(!connection.getAutoCommit()){
-						connection.commit();
-					}
-					connection.close();
-				}
-			} catch (Exception e) {
-				Logger.debug(StartupTasksExecutor.class, "Exception: "
-						+ e.getMessage(), e);
-				throw new DotDataException(
-						"Exception finishing upgrade tasks: " + e.getMessage(),
-						e);
-			} finally {
-
-				CloseUtils.closeQuietly(connection);
-			}
-			Logger.info(this, "Finishing upgrade tasks.");
-		}
-
-	}
+    
+	private boolean insureDbVersionTable() {
+
+        try {
+            currentDbVersion();
+            return true;
+
+        } catch (Exception e) {
+            return createDbVersionTable();
+        }
+
+    }
+    
+    
+    private int currentDbVersion() {
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()) {
+            Map<String,Object> results =  new DotConnect().setSQL(select).loadObjectResults().get(0);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'results' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)